feat: add feature flag for baseline grid (#2)

csshugs opened the pull request:
because:
- not all projects require a global baseline grid
- the compiling of a line-height declaration each time a font-size
is declared may not be worth it in all cases

this commit:
- adds the ability to set the `$FLUIDMS-GLOBAL-BASELINE` to `false`
to globally disable the baseline grid

csshugs committed:
feat: add feature flag for baseline grid (#2)

because:
- not all projects require a global baseline grid
- the compiling of a line-height declaration each time a font-size
is declared may not be worth it in all cases

this commit:
- adds the ability to set the `$FLUIDMS-GLOBAL-BASELINE` to `false`
to globally disable the baseline grid

fuhlig commented on src/settings/_settings.config.scss:
```diff
@@ -9,10 +9,13 @@
-/// @type `rem` value
+/// @type `rem` value or `false`
 ///
 /// @example scss - Sets the size of the baseline grid to `0.2rem`
```
[see related comment](https://github.com/csshugs/FluidMS/pull/19#discussion_r355116520)

fuhlig commented on src/generic/_generic.ms-custom-properties.scss:
```diff
@@ -9,7 +9,9 @@
         --ms#{$scale}-ratio:       #{fluidms-pow(map-get($FLUIDMS-CONFIG, ratio), $scale)};
-        --ms#{$scale}-line-height: #{fluidms-line-height($scale, map-get($FLUIDMS-CONFIG, ratio))};
+        @if $FLUIDMS-GLOBAL-BASELINE {
+            --ms#{$scale}-line-height: #{fluidms-line-height($scale, map-get($FLUIDMS-CONFIG, ratio))};
```
why does the baseline have a unit (`rem`) when applied to `line-height`?

csshugs replied:
The baseline in general needs to have a unit, because it is not tied to any font-size. As we want it all to be fluid, the unit needs to be `rem`. 

Since we are managing the line-heights via CSS custom properties, we cannot use unit-less line-heights, because ... well ... it’s very complicated to calculate (if even possible), especially when using the `$line-height-modifier` argument. But more importantly, when a global baseline grid is needed, I see no problem in specifically defining the line-heights in `rem`s, as the baseline unit itself is defined in `rem`s, too. Don’t forget, that there can’t be one single unit-less line-height **for all font-sizes** anyway, when the global baseline grid is active. 

It seems simpler and more clear this way.

fuhlig commented on src/tools/_tools.font-size.scss:
```diff
@@ -95,16 +102,23 @@
+                }
+                line-height: $_line-height $important;
+            }
+            @if (not $FLUIDMS-GLOBAL-BASELINE) {
```
duplication (l91-95)

fuhlig replied:
there is still a duplication or am I missing something?

csshugs replied:
@fuhlig How would you eliminate that duplication?

fuhlig commented on src/tools/_tools.line-height.scss:
```diff
@@ -20,15 +20,18 @@
+        $_default-line-height: (fluidms-pow($ratio, $scale)) * $FLUIDMS-GLOBAL-LINE-HEIGHT;
+        // Get the “offset” of the vertical grid.
+        $_remainder: $_default-line-height;
+        @while $_remainder >= fluidms-strip-unit($FLUIDMS-GLOBAL-BASELINE) {
```
consider putting conditions in parentheses for improved readability

```scss
@directive (condition && condition)
vs.
@directive condition && condition
```

csshugs committed:
docs: remove SassDoc example (#2)

because:
- we don’t need to show an example for every little detail

this commit:
- removes purposeless SassDoc comment which outputs an code
example

csshugs committed:
refactor: eliminate code duplication (#2)

csshugs committed:
style: improve readability (#2)

fuhlig commented on src/tools/_tools.font-size.scss:
```diff
@@ -77,15 +77,22 @@
+                }
+                line-height: $_line-height $important;
+            }
+            @if (not $FLUIDMS-GLOBAL-BASELINE) {
```
consider using `@else` instead of `@if condition...@if (not condition)`

csshugs committed:
style: reverse if-condition (#2)
